### ISSUE TITLE
Doesn't build fonts folder properly? 

### ISSUE THREAD

**Issue by bobbyziom** [ngbp/ngbp#355]
Can't find fonts folder when built.

Could it have something to do with this grunt warning?

WARN [watcher]: Pattern "/Users/[CurrentUser]/Documents/[Folder]/my-project-name/src/*_/_.coffee" does not match any file.

localhost/:1 GET http://localhost:8000/fonts/glyphicons-halflings-regular.woff 
localhost/:1 GET http://localhost:8000/fonts/glyphicons-halflings-regular.ttf 404 (File not found)

(when using ui.boostrap timepicker)

**Comment by joshdmiller** [comment]
As Font Awesome is part of the boilerplate, the glyphicons are not included. But you should be able to add them manually as vendor assets.

**Comment by bobbyziom** [comment]
Thanks for the reply!

I can see that these are already included in the fonts folder of vendor->bootstrap. How could I get those to build in local:8000/fonts, which angular ui is looking for? If I put them to assets they will build in the asset folder, right? 

![screen shot 2015-02-14 at 13 10 13](https://cloud.githubusercontent.com/assets/5149004/6199550/f4c77822-b44a-11e4-9ed7-69c8652fa5d3.png)

**Comment by joshdmiller** [comment]
Yes, that's true - they'll be in `/assets`. However, it's not angular ui that is requiring them, but twbs, right? And there's a [less variable that you can overwrite](https://github.com/twbs/bootstrap/blob/master/less/variables.less#L79). Let me know how it goes.

**Comment by bobbyziom** [comment]
Okay, I'm a beginner in javascript frameworks and bower. I was not aware that I had to edit the build config file manually after installing packages from bower. It turned out that overwriting the variable in the less file did actually work, thanks! 
I added the vendor/bootstrap/fonts/*\* path to the js vendor file build config, and changed the path in the variables.less to be /vendor/bootstrap/fonts. I should maybe add them in the assets part of the vendor build. But it seems like it's creating an empty folder and placing the files directly into the assets folder. But yeah, it works now! Again - thanks for your help! :)

**Comment by tristobal** [comment]
I have the same problem when i instanciated the datepicker of Ui Bootstrap (included in this boilerplate) because this use glyphicons.

I solved it this way:

build.config.js

```
  vendor_files: {
    js: [
        // angular dependencies
    ],
    css: [
    ],
    assets: [
    ],
    fonts: [
        'vendor/bootstrap/fonts/glyphicons-halflings-regular.eot',
        'vendor/bootstrap/fonts/glyphicons-halflings-regular.svg',
        'vendor/bootstrap/fonts/glyphicons-halflings-regular.ttf',
        'vendor/bootstrap/fonts/glyphicons-halflings-regular.woff'
    ]
  },
```

Gruntfile.js

```
  //inside of copy: { }
  build_fonts: {
      files: [
          {
              src: [ '<%= vendor_files.fonts %>' ],
              dest: '<%= build_dir %>/assets/fonts',
              cwd: '.',
              expand: true,
              flatten: true,
              filter: 'isFile'                        
          }
      ]
  },

//add 'copy:build_fonts'
grunt.registerTask( 'build', [
    'clean', 'html2js', 'jshint', 'coffeelint', 'coffee', 'less:build',
    'concat:build_css', 'copy:build_app_assets', 'copy:build_vendor_assets',
    'copy:build_appjs', 'copy:build_vendorjs', 'copy:build_vendorcss', 'copy:build_fonts',
    'index:build', 'karmaconfig', 'karma:continuous'
]);
```

variables.less

```
@icon-font-path: '../assets/fonts/';
```

**Comment by ghost** [comment]
Hello! I have an issue with Font-Awesome icons. I've set up FA files in my project's vendor folder and included font-awesome.less (configurations, font-face declaration and the general imports) into main.less. the build process successfully import FA less codes, but the fonts (eot,ttf,...etc) have not been copied into build/assets/fonts and I can't see the correct result on my built site. I've followed the solution provided by @tristobal  above, but nothing changed. Anyone can give me a hint what I'm missed here? Or is there another way to set up the full set of FA icons?
Thank you very much!

Folder structure

```
- vendor/
   ...
  - font-awesome/
    - fonts/
       fontawesome-webfont.eot
       ...
    - less/
       variables.less
       ...
    font-awesome.less

```

**Comment by manichandra** [comment]
I was able to achieve it with the help of @tristobal solution . Thanks @tristobal .
Here is my solution if it helps . 

build.config.js

```
vendor_files: {
js: [ ],
css: [ ],
css_font: [
  'vendor/angular-ui-grid/ui-grid.ttf',
  'vendor/angular-ui-grid/ui-grid.svg',
  'vendor/angular-ui-grid/ui-grid.woff'
],
less_font: [
      /*'vendor/font-awesome/font/!*',*/
      'vendor/bootstrap/fonts/*'
],
assets: [
]
```

gruntfile.js 

inside copy function

```
build_vendor_css_font: {
    files: [
      {
        src: [ '<%= vendor_files.css_font %>' ],
        dest: '<%= build_dir %>/',
        cwd: '.',
        expand: true
      }
    ]
  },
  build_vendor_less_font: {
    files: [
      {
        src: [ '<%= vendor_files.less_font %>' ],
        dest: '<%= build_dir %>/fonts/',
        flatten: true,
        cwd: '.',
        expand: true
      }
    ]
  },
```

build Task
      grunt.registerTask( 'build', ['clean', 'html2js', 'jshint', 'coffeelint', 'coffee', 'less:build',               'copy:build_app_assets',  copy:build_vendor_assets','copy:build_appjs', 'copy:build_vendorjs', 'copy:build_vendorcss','copy:build_vendor_css_font','copy:build_vendor_less_font','concat:build_css', 'index:build', 'karmaconfig','karma:continuous' ]);
